src/validator-info.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+
+import * as BufferLayout from 'buffer-layout';
+import * as Layout from './layout';
+import {PublicKey} from './publickey';
+import {Account} from './account';
+import * as shortvec from './util/shortvec-encoding';
+import {struct} from 'superstruct';
+
+const VALIDATOR_INFO_KEY = new PublicKey('Va1idator1nfo111111111111111111111111111111');
+
+/**
+ * @private
+ */
+type ConfigKey = {|
+  publicKey: PublicKey,
+  isSigner: boolean,
+|};
+
+/**
+ * @private
+ */
+type Info = {|
+  name: string,
+  website?: string,
+  details?: string,
+  keybaseId?: string,
+|};
+
+const InfoString = struct({
+  name: 'string',
+  website: 'string?',
+  details: 'string?',
+  keybaseId: 'string?',
+});
+
+/**
+ * ValidatorInfo class
+ */
+export class ValidatorInfo {
+  key: PublicKey;
+  info: Info;
+
+  /**
+   * Construct a valid ValidatorInfo
+   */
+  constructor(key: PublicKey, info: Info) {
+    this.key = key;
+    this.info = info;
+  }
+
+  /**
+   * Deserialize ValidatorInfo from the config account data.
+   *
+   * @return ValidatorInfo or null if info was not found
+   */
+  static fromConfigData(buffer: Buffer): ?ValidatorInfo {
+    const PUBKEY_LENGTH = 32;
+
+    let byteArray = [...buffer];
+    const configKeyCount = shortvec.decodeLength(byteArray);
+    let configKeys: Array<ConfigKey> = [];
+    for (let i = 0; i < configKeyCount; i++) {
```

Comment:
How about we change this loop to only look for the first 2 keys, we don't really care about if any other keys are present.